# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditor.h
# Commit: b9a3b6f27a8d
# Full Hash: b9a3b6f27a8d20c0b3ce38dc1a8f07cc4015afac
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-08-04 04:34:13
# Regressor Bug: 1774704
# File Overlap Count: 1
# Description:
#   Bug 1774704 - part 1: Make `HTMLEditor::EnsureCaretInBlockElement` only computes new caret point in the given element r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152963
# ==============================================================================

diff -r d554ca7850eb -r b9a3b6f27a8d editor/libeditor/HTMLEditor.h
--- a/editor/libeditor/HTMLEditor.h	Thu Aug 04 00:21:53 2022 +0000
+++ b/editor/libeditor/HTMLEditor.h	Thu Aug 04 00:21:53 2022 +0000
@@ -2613,14 +2613,6 @@
   void SetSelectionInterlinePosition();
 
   /**
-   * EnsureSelectionInBlockElement() may move caret into aElement or its
-   * parent block if caret is outside of them.  Don't call this when
-   * `Selection` is not collapsed.
-   */
-  [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult
-  EnsureCaretInBlockElement(dom::Element& aElement);
-
-  /**
    * Called by `HTMLEditor::OnEndHandlingTopLevelEditSubAction()`.  This may
    * adjust Selection, remove unnecessary empty nodes, create `<br>` elements
    * if needed, etc.
